packages/select/src/utils/getLGSelectTestUtils/getLGSelectTestUtils.ts

```diff
@@ -0,0 +1,160 @@
+   */
+  const selectTrigger = queryByQuerySelector<HTMLButtonElement>(
+    element,
+    'button',
+  ) as HTMLButtonElement;
+
+  /**
+   * Queries the `element` for the error message element. Will return `null` if the error message is not found.
+   */
+  const errorMessage = queryByQuerySelector<HTMLElement>(
+    element,
+    `[data-lgid="${LGID_SELECT_ERROR_MESSAGE}"]`,
+  );
+
+  /**
+   * Queries the `element` for aria-disabled.
+   */
+  const isInputDisabled = () => {
+    const ariaDisabled = queryByQuerySelector<HTMLElement>(
+      element,
+      '[aria-disabled="true"]',
+    );
+
+    return !!ariaDisabled;
+  };
+
+  /**
+   * Returns the `Select` value.
+   */
+  const getSelectValue = () => {
+    return selectTrigger.textContent || '';
+  };
+
+  /**
+   * Queries the `element` for the warning Icon.
+   */
+  const isError = () => {
+    const warningIcon = queryByQuerySelector<SVGElement>(
+      element,
+      'svg[title="Error"]',
+    );
+
+    return !!warningIcon;
+  };
+
+  // We cannot query within the select element because if the popover is using a portal, the element will render outside the select element
+  const getPopover = () =>
+    queryByQuerySelector<HTMLDivElement>(
+      document.body,
+      '[data-lgid="lg-select-popover"]',
+    );
+
+  const getAllOptions = (): Array<HTMLLIElement> => {
+    // only one select popover should be open at a time
+    const popover = getPopover();
+
+    if (!popover)
+      throw new Error(
+        'Unable to find an element by: [data-lgid="lg-select-popover"]',
+      );
+
+    // Find all options
+    const allOptions =
+      popover.querySelectorAll<HTMLLIElement>('[role="option"]');
+
+    // unlikley to happen since the select will not render without options
+    if (!allOptions.length)
+      throw new Error('Unable to find any `Select` options.');
+
+    return Array.from(allOptions);
+  };
+
+  const getOptionByValue = (value: string) => {
+    if (!value) throw new Error('Value cannot be empty');
+
+    const allOptions = getAllOptions();
+
+    // Find the option with the value
+    const option = allOptions.find(node => node.textContent === value);
+
+    if (!option) return null;
+
+    return option;
+  };
+
+  const clickOption = (value: string) => {
+    if (!value) throw new Error('Value cannot be empty');
+
+    const option = getOptionByValue(value);
+
+    if (!option)
+      throw new Error(`Unable to find option with the value '${value}'.`);
+
+    // Click option
+    option.click();
+  };
+
+  const clickTrigger = () => {
+    selectTrigger.click();
+  };
+
+  return {
+    elements: {
+      getLabel: () => label,
+      getDescription: () => description,
+      getInput: () => selectTrigger,
+      getErrorMessage: () => errorMessage,
+      getOptions: () => getAllOptions(),
+      getOptionByValue: (value: string) => getOptionByValue(value),
+      getPopover: () => getPopover(),
+    },
+    utils: {
```

Comment:
Sitting with this more, I feel like this should just be one big return object instead of `elements` & `utils`. The line can get fuzzy

Reply:
i can see that, react-testing-lib does return everything in one big object

Reply:
I spoke with @bruugey, and we're ok with combining these into one big object.